src/main/java/io/vertx/openapi/validation/Parameter.java

```diff
@@ -18,13 +18,22 @@
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 
+import java.util.function.Supplier;
+
 interface Parameter {
   /**
    * @return null if value is not a {@link String}, otherwise it returns value
    */
   @Nullable
   default String getString() {
-    return isString() ? (String) get() : null;
+    return getStringOrDefault(() -> null);
+  }
+
+  /**
+   * @return the default supplied by a {@link java.util.function.Supplier} if value is not a {@link String} or is null, otherwise it returns the value.
+   */
+  default String getStringOrDefault(Supplier<String> defaultValue) {
```

Comment:
The reason I chose to do ```getStringOrDefault(Supplier)``` is because this is a very special use case of Parameter itself. I think you (should) only use this function specifically when the Parameter contains a default. For all other use cases you should use ```getString()```.

Parameter is required in the spec -> use ```getString()```
Parameter has a default in the spec -> use ```getStringOrDefault(Supplier)```
Parameter is not required, and also doesn't have a default -> use ```getString()``` (and you would check ```isNull()``` still)

The spec you are using I think will change what function you would call specifically and I wanted that highlighted.

Ultimately either or is OK with me 😃

Reply:
Maybe there is a misunderstanding. I don't mean get rid of getString(). I mean just overload the getString() method with a getString(Supplier).

Reply:
Nono sorry I don't mean get rid of ```getString()``` but at least for me when it comes to choosing which function to use for my use case. I think having ```getStringOrDefault()``` would be better as this function is really only meant for Parameters that have defaults.

Reply:
Any other thoughts on this @pk-work ?

Reply:
Vert.x JsonObject does use this pattern with default values, it is a simple overload.

```java
public String getString(String key, String def) {
   ...
}
```

Reply:
I think we could use the same

Reply:
Sounds good I've updated to overload and removed 'orDefault'!